packages/react-components/source/react/internal/action-menu-list/ActionMenuList.js

```diff
@@ -247,15 +246,21 @@ class ActionMenuList extends Component {
           {...rest}
         >
           {actions.map(
-            ({ id: actionId, label, icon, svg, onClick, ...other }, index) => (
+            (
+              { id: actionId, label, icon, svg, onClick, disabled, ...other },
+              index,
+            ) => (
               <ActionMenuListItem
                 id={getActionId(id, actionId)}
                 key={actionId}
                 focused={index === focusedIndex}
                 icon={icon}
                 svg={svg}
                 onMouseEnter={() => onMouseEnterItem(index)}
-                onClick={e => executeAction(e, onClick, actionId)}
+                onClick={e =>
+                  disabled ? () => {} : executeAction(e, onClick, actionId)
```

Comment:
Would it be better to do something like this?

```suggestion
                  !disabled && executeAction(e, onClick, actionId)
```